hypothesis/lms
# Add logic to save email preferences in EmailNotificationsApp

Depends on https://github.com/hypothesis/lms/pull/5900, and is part of https://github.com/hypothesis/lms/issues/5812

This PR adds logic to save data loaded in `EmailPreferences`.

As agreed in https://hypothes-is.slack.com/archives/C1MA4E9B9/p1701872286024709 and https://hypothes-is.slack.com/archives/C1MA4E9B9/p1701875517541549, this PR implements the saving logic in a way that we don't need to change the BE, as there are still some open questions and we don't want to close any doors yet, and by keeping the synchronous form submit.

On follow-up PRs I will implement:
* Displaying some kind of toast message or caption with the saving result
* Improving the overall app UI.

> [!NOTE]
> This PR is easier to review [ignoring whitespaces](https://github.com/hypothesis/lms/pull/5902/files?w=1)

### Implementation decisions

~For now we just want a form to be synchronously called, but the `onSave` prop from `EmailNotificationsPreferences` exposes both the submit event and currently selected days.~

~With that, a consumer would be able to call event.preventDefault() and then handle selected days as desired.~

~That will be useful if we decide to display these in other contexts.~

UPDATE: We finally decided to simplify and focus on the synchronous submit use case. Let's make any modifications to this when/if needed.

### Testing steps

Follow the testing steps described in https://github.com/hypothesis/lms/pull/5900, but now, the save button should actually allow you to save the selected days.

There's one consideration, which is that refresing the page will result in a Pyramid 404 error, because once the `/email/preferences` backend route is loaded, our frontend does a client-side routing to `/app/email-notifications`. This route is not known by the backend, hence the 404.

I will try to address that on a follow-up PR. For now, if you want to check the preferences were properly saved, reopen the URL generated on the first step.
